packages/contracts/contracts/test/Colosseum.t.sol

```diff
@@ -1553,3 +1553,179 @@ contract Colosseum_ValidatorSystemUpgrade_Test is Colosseum_Initializer {
+            _maxUnbond: maxUnbond,
+            _roundDuration: roundDuration,
+            _terminateOutputIndex: terminateOutputIndex
+        });
+        vm.prank(multisig);
+        Proxy(payable(address(pool))).upgradeTo(address(poolImpl));
+
+        // Submit outputs until ValidatorPool is terminated
+        vm.prank(trusted);
+        pool.deposit{ value: trusted.balance }();
+        for (uint256 i; i <= terminateOutputIndex; i++) {
+            _submitL2OutputV1();
+        }
+
+        // Only trusted validator can submit the first output with ValidatorManager
+        _registerValidator(trusted, minActivateAmount);
+
+        // Submit invalid output as asserter
+        uint256 nextBlockNumber = oracle.nextBlockNumber();
+        warpToSubmitTime();
+        vm.prank(valMgr.nextValidator());
+        oracle.submitL2Output(keccak256(abi.encode()), nextBlockNumber, 0, 0);
+
+        // To create challenge, challenger also registers validator
+        _registerValidator(challenger, minActivateAmount);
+
+        targetOutputIndex = oracle.latestOutputIndex();
+    }
+
+    function _getOutputRoot(address sender, uint256 blockNumber) private view returns (bytes32) {
+        uint256 targetBlockNumber = ColosseumTestData.INVALID_BLOCK_NUMBER;
+        if (blockNumber == targetBlockNumber - 1) {
+            return ColosseumTestData.PREV_OUTPUT_ROOT;
+        }
+
+        // If asserter, wrong output after targetBlockNumber
+        if (sender == trusted) {
+            if (blockNumber < targetBlockNumber - 1) {
+                return keccak256(abi.encode(blockNumber));
+            } else {
+                return keccak256(abi.encode());
+            }
+        }
+
+        // If challenger, correct output always
+        if (blockNumber == targetBlockNumber) {
+            return ColosseumTestData.TARGET_OUTPUT_ROOT;
+        } else {
+            return keccak256(abi.encode(blockNumber));
+        }
+    }
+
+    function _newSegments(
+        address sender,
+        uint8 turn,
+        uint256 segStart,
+        uint256 segSize
+    ) private view returns (bytes32[] memory) {
+        uint256 segLen = colosseum.segmentsLengths(turn - 1);
+
+        bytes32[] memory arr = new bytes32[](segLen);
+
+        for (uint256 i = 0; i < segLen; i++) {
+            uint256 n = segStart + i * (segSize / (segLen - 1));
+            arr[i] = _getOutputRoot(sender, n);
+        }
+
+        return arr;
+    }
+
+    function _getFirstSegments() private view returns (bytes32[] memory) {
+        Types.CheckpointOutput memory targetOutput = oracle.getL2Output(targetOutputIndex);
+        uint256 end = targetOutput.l2BlockNumber;
+        uint256 start = end - oracle.SUBMISSION_INTERVAL();
+
+        bytes32[] memory segments = _newSegments(challenger, 1, start, end - start);
+
+        return segments;
+    }
+
+    function test_createChallenge_succeeds() public {
+        bytes32[] memory segments = _getFirstSegments();
```

Comment:
Maybe we can declare this at `setUp` function, so that we can use it as a global variable. Hdyt?

Reply:
Yes, I want to unify the common functions into an initializer contract, but there are dependencies on the setup function and member variables of the contract.